CMakeLists.txt

```diff
@@ -77,6 +77,7 @@ kafka
 lfs
 lpeg
 lsb
+moz_generic
```

Comment:
I've switched the base to dev.

See [this google doc](https://docs.google.com/document/d/1PqiF1rF2fCk_kQuGSwGwildDf4Crg9MJTY44E6N5DSk/edit?ts=5910c4cf) in addition to [these](https://bugzilla.mozilla.org/show_bug.cgi?id=1368197) [bugs](https://bugzilla.mozilla.org/show_bug.cgi?id=1368196) for some specs.

Generic ingestion is essentially telemetry without all the edge case logic. The theory is that we could support other input and output schemas eventually, it's just that we don't have any use cases that aren't json at the moment.

While we could implement this as an extension of the telemetry decoder, I think it would result in more complicated code with a high chance of causing unexpected regressions. I would prefer to maintain a simpler, separate decoder that cannot affect telemetry processing to changing the current decoder. @mreid-moz may have an opinion on this as well.

If we do want to maintain a single decoder, I would be most comfortable with some kind of top-level flag applied to decoders performing generic ingestion that makes the branching logic very apparent while still allowing us to avoid code duplication.